### PR TITLE
Deepcopy tests for slices and manual DeepCopy methods

### PR DESCRIPTION
**Pull request by thockin**
Amazing the things you find when you test.

### CODE REVIEW

**Comment by wojtek-t** [comment]
One minor question - feel free to self-lgtm it.

**Comment by thockin** [comment]
What was the question?  I think you forgot to ask it :)

**Comment by wojtek-t** [comment]
Sorry - I was on vacation. You shouldn't have wait for me with this PR. And now I clicked 'Comment' finally.

File: examples/deepcopy-gen/test/wholepkg/deepcopy_generated.go [hunk 1]
```diff
@@ -54,6 +60,15 @@ func DeepCopy_wholepkg_ManualStruct(in interface{}, out interface{}, c *conversi
 	}
 }
 
+func DeepCopy_wholepkg_ManualStruct_Alias(in interface{}, out interface{}, c *conversion.Cloner) error {
+	{
```
Comment (thockin) [inline examples/deepcopy-gen/test/wholepkg/deepcopy_generated.go]: good question.  It's not in this PR, but I'll look for it.

**Comment by thockin** [comment]
I'm gonna merge this and hunt that issue in a new PR

File: examples/deepcopy-gen/test/wholepkg/deepcopy_generated.go (hunk 1 above)
Comment (thockin) [inline examples/deepcopy-gen/test/wholepkg/deepcopy_generated.go]: This is needed to do the `in := in.(*cast)` stuff.  without the extra nesting, it doesn't compile.  It has been there for a long time.

**Pull request merged**